Src/FluentAssertions/Formatting/EnumerableValueFormatter.cs

```diff
@@ -49,34 +45,27 @@ public void Format(object value, FormattedObjectGraph formattedGraph, Formatting
             {
                 using IDisposable _ = formattedGraph.WithIndentation();
                 string moreItemsMessage = value is ICollection c ? $"…{c.Count - MaxItems} more…" : "…more…";
-                AddLineOrFragment(formattedGraph, startCount, moreItemsMessage);
+                iteratorGraph.AddLineOrFragment(moreItemsMessage);
+            }
+
+            if (iterator.IsFirst)
```

Comment:
❓ Why did you move this down?

Reply:
Because we cannot know whether or not the enumerable will take up more than one line of output until we have formatted the first item.  So we format the first item, then go back and insert the enumerable's opening brace in the correct place depending on whether that first item was all on one line or not.

I did consider formatting the item into a throw-away graph to detect if it was multi-line or not, but that proved tricky as the list could contain objects that themselves contained lists.

Reply:
> Because we cannot know whether or not the enumerable will take up more than one line of output until we have formatted the first item. So we format the first item, then go back and insert the enumerable's opening brace in the correct place depending on whether that first item was all on one line or not.

I guess this belongs in the code as in-line comments